File: apps/api/src/main/java/br/com/agendaplatform/waitlist/application/WaitlistSummary.java
package br.com.agendaplatform.waitlist.application;

import java.time.Instant;
import java.time.LocalDate;
import java.time.LocalTime;
import java.util.UUID;

public record WaitlistSummary(
        UUID id,
        UUID serviceId,
        String clientName,
        String serviceName,
        LocalDate preferredStartDate,
        LocalDate preferredEndDate,
        LocalTime preferredStartTime,
        LocalTime preferredEndTime,
        String priority,
        Instant expiresAt,
        String status,
        UUID appointmentId) {
}

File: apps/admin/src/app/core/waitlist/waitlist-summary.ts
export interface WaitlistSummary {
  readonly id: string;
  readonly serviceId: string;
  readonly clientName: string;
  readonly serviceName: string;
  readonly preferredStartDate: string;
  readonly preferredEndDate: string;
  readonly preferredStartTime: string;
  readonly preferredEndTime: string;
  readonly priority: 'LOW' | 'NORMAL' | 'HIGH';
  readonly expiresAt: string;
  readonly status: 'WAITING' | 'EXPIRED' | 'CONVERTED' | 'CANCELLED';
  readonly appointmentId: string | null;
}
